### PR TITLE
Reorganize "What is ROCm?" page

### PR DESCRIPTION
**Pull request by peterjunpark**
This PR adds an image to illustrate the list of projects in the ROCm stack and aims to improve the readability of the table of ROCm projects on the "What is ROCm?" page.

- Added diagram of ROCm SW stack
- Split up ROCm project table by project type

### CODE REVIEW

File: docs/what-is-rocm.rst [hunk 1]
```diff
@@ -26,57 +29,105 @@ ROCm project list
 ROCm consists of the following projects. For information on the license associated with each project,
 see :doc:`ROCm licensing <./about/license>`.
 
+Compilers
```
Comment (yhuiYH) [inline docs/what-is-rocm.rst:32]: Can we organize the tables, just like the graphic?  From Top -> Bottom, so it's move intuitive for users to find what they're looking for?

So Compilers move between Tools and Runtime.  

Tools
Compilers
Runtime

File: docs/what-is-rocm.rst [hunk 2]
```diff
@@ -26,57 +29,105 @@ ROCm project list
...
+-----------------------------------------------
+
+.. csv-table::
+  :header: "Project", "Description"
+
+  "`AOMP <https://github.com/ROCm/aomp/>`_", "Scripted build of `LLVM <https://github.com/ROCm/llvm-project>`_ and supporting software"
+  "`FLANG <https://github.com/ROCm/flang/>`_", "An out-of-tree Fortran compiler targeting LLVM"
+  "`hipCC <https://github.com/ROCm/HIPCC>`_ ", "Compiler driver utility that calls Clang or NVCC and passes the appropriate include and library options for the target compiler and HIP infrastructure"
+  ":doc:`HIPIFY <hipify:index>`", "Translates CUDA source code into portable HIP C++"
```
Comment (yhuiYH) [inline docs/what-is-rocm.rst:41]: Can we move HIPIFY to Tools (it probably wasn't classified correctly before)

File: docs/what-is-rocm.rst [hunk 3]
```diff
@@ -26,57 +29,105 @@ ROCm project list
...
+  "`LLVM (amdclang) <https://github.com/ROCm/llvm-project>`_ ", "Toolkit for the construction of highly optimized compilers, optimizers, and runtime environments"
+
+Libraries
+-----------------------------------------------
+
+AI/ML
```
Comment (yhuiYH) [inline docs/what-is-rocm.rst:47]: Can we match the graphics now?  Machine Learning & Computer Vision

File: docs/what-is-rocm.rst [hunk 4]
```diff
@@ -10,6 +10,9 @@ ROCm is an open-source stack, composed primarily of open-source software, design
 graphics processing unit (GPU) computation. ROCm consists of a collection of drivers, development
 tools, and APIs that enable GPU programming from low-level kernel to end-user applications.
 
+.. figure:: data/rocm-sw-stack-chart.png
```
Comment (yhuiYH) [inline docs/what-is-rocm.rst:13]: I see the filename is causing spelling errors.  Perhaps we:
1) Use the JPG
2) filename change to rocm-software-stack-6_1_0.jpg

Comment (peterjunpark) [reply docs/what-is-rocm.rst]: "jpg" is also flagged by the linter 😭

**Review by Rmalavally — COMMENTED** [review]
Looks good, @peterjunpark. Nice job!

**Review by Rmalavally — COMMENTED** [review]
@peterjunpark @yhuiYH We seem to be missing programming models in the image.

File: docs/what-is-rocm.rst [hunk 5]
```diff
@@ -20,63 +25,111 @@ frameworks, such as PyTorch and TensorFlow.
...
+  ":doc:`rocALUTION <rocalution:index>`", "Sparse linear algebra library for exploring fine-grained parallelism on ROCm runtime and toolchains"
+  ":doc:`rocBLAS <rocblas:index>`", "BLAS implementation (in the HIP programming language) on the ROCm runtime and toolchains"
+  ":doc:`rocFFT <rocfft:index>`", "Software library for computing fast Fourier transforms (FFTs) written in HIP"
+  ":doc:`rocRAND <rocrand:index>`", "Provides functions that generate pseudorandom and quasirandom numbers"
+  ":doc:`rocSOLVER <rocsolver:index>`", "An implementation of LAPACK routines on ROCm software, implemented in the HIP programming language and optimized for AMD's latest discrete GPUs"
+  ":doc:`rocSPARSE <rocsparse:index>`", "Exposes a common interface that provides BLAS for sparse computation implemented on ROCm runtime and toolchains (in the HIP programming language)"
+  ":doc:`rocWMMA <rocwmma:index>`", "C++ library for accelerating mixed-precision matrix multiply-accumulate (MMA) operations"
+  "`Tensile <https://github.com/ROCm/Tensile>`_ ", "Creates benchmark-driven backend libraries for GEMMs, GEMM-like problems, and general N-dimensional tensor contractions"
+
+C++ primitives
```
Comment (yhuiYH) [inline docs/what-is-rocm.rst:82]:
```suggestion
Primitives
```

Should we change this title to align with the graphic? @lpaoletti

File: docs/what-is-rocm.rst (hunk 3 above)
Comment (peterjunpark) [reply docs/what-is-rocm.rst]: Fixed

File: docs/what-is-rocm.rst (hunk 4 above)
Comment (peterjunpark) [reply docs/what-is-rocm.rst]: Fixed

File: docs/what-is-rocm.rst (hunk 5 above)
Comment (peterjunpark) [reply docs/what-is-rocm.rst]: Fixed

**Review by yhuiYH — APPROVED** [review]
Looks good to me

**Review by samjwu — APPROVED** [review]

**Pull request merged**